docs/reference/hooks.md

```diff
@@ -0,0 +1,154 @@
+{: .note } 
+
+### Action file Webhook properties
+
+|Property          |Description                                            |Required |Default Value
+|------------------|-------------------------------------------------------|---------|--------------------------------------|
+|url               |The URL address of the request                         |true     |
+|timeout           |Time to wait for response before failing the hook      |false    | 1m 
+|query_params      |List of query params that will be added to the request |false    |
+
+Example:
+```yaml
+...
+hooks:
+  - id: prevent_user_columns
+    type: webhook
+    description: Ensure no user_* columns under public/
+    properties:
+      url: "http://<host:port>/webhooks/schema"
+      timeout: 1m30s
+      query_params:
+        disallow: ["user_", "private_"]
+        prefix: public/
+...
+```
+
+### Request body schema
+A JSON object with the following fields:
+
+|Field             |Description                                                                          |Type  |Example                  |
+|------------------|-------------------------------------------------------------------------------------|------|-------------------------|
+|EventType         |Type of the event that triggered the `Action`                                        |string|pre_commit               |
+|EventTime         |Time of the event that triggered the `Action` (RFC3339)                              |string|2006-01-02T15:04:05Z07:00|
+|ActionName        |Containing `Hook` Action's Name                                                        |string|                         |
+|HookID            |ID of the `Hook`                                                                       |string|                         |
+|RepositoryID      |ID of the Repository                                                                 |string|
+|BranchID          |ID of the Branch                                                                     |string|
+|SourceRef         |Reference to the source that triggered the event (source Branch for commit or merge) |string|
+|CommitMessage     |The message for the commit (or merge) that is taking place                           |string|
+|Committer         |Name of the committer                                                                |string|
+|CommitMetadata    |The metadata for the commit that is taking place                                     |string|
+
+Example:
+```json
+{
+  "event_type": "pre-merge",
+  "event_time": "2021-02-28T14:03:31Z",
+  "action_name": "test action",
+  "hook_id": "prevent_user_columns",
+  "repository_id": "repo1",
+  "branch_id": "feature-1",
+  "source_ref": "feature-1",
+  "commit_message": "merge commit message",
+  "committer": "committer",
+  "commit_metadata": {
+    "key": "value"
+  }
+}
+```
```

Comment:
Missing a link to the API reference and the CLI hooks sub-command. 
Also, I'd add another paragraph at the beginning explaining some of the use cases for this (you can take example from the lakeFS-hooks repo)

Reply:
Done.